Z2A SwiftUI/NSCoffee/NSCoffee/Models/Basket.swift

```diff
@@ -22,16 +22,18 @@ class Basket: ObservableObject {
         orders.reduce(into: 0.0) { $0 += $1.totalPrice }
     }
 
-    func add(_ drink: any Drink) {
-        if let orderIndex = orders.firstIndex(where: { $0.drink.name == drink.name }) {
+    func add(_ order: Order) {
```

Comment:
I like that you are treating drinks with different extras as different orders. Will implement that in UIKit too.